File: server/src/main/java/project/server/Entities/DTO/CourseDTO.java
package project.server.Entities.DTO;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class CourseDTO {
    
    private String name;
    private String description;   
    private Long subject ;
    private Long creator ;
    
}

File: client/src/app/entities/DTO/courseDTO.ts
import { Quizz } from "../quizz";
import { Tutor } from "../tutor";

export interface CourseDTO {
    id: number;
    name: string;
    description: string;
    imagePath: string;
    creator: Tutor;
    subject: any;
    quizz: Quizz;
    rating: number;
    duration: number;
}